src/Query.php

```diff
@@ -352,9 +360,40 @@ public function getModels($columns = ['*'])
 
 		$items = $this->db->setQuery($this->query)->loadAssocList();
 
+		$this->resetQuery();
+
 		return $this->hydrate($items)->all();
 	}
 
+	/**
+	 * Function to get the raw attributes for a raw in the table.
+	 *
+	 * @param   mixed  $id      primary key, if there is no key, then this is used for a new item, therefore select last
+	 * @param   array  $columns columns to be selected in query
+	 *
+	 * @internal
+	 * @return mixed
+	 */
+	public function selectRaw($id, $columns = ['*'])
```

Comment:
This is needed in order to load a new row in the current instance of the Model. But, as for now, the load() function is in the EntityTableTrait. Since I started integrating also the com_content, it seems that this is the most used usecase, so maybe we should move it into the Entity. https://github.com/joomla-projects/gsoc18_webservices/pull/66/files#diff-de77090714147c7dac6a956be8d275d9R78